### DISCUSSION TITLE
How to pause and resume mitmproxy? Shutdown does not free up the port.

### DISCUSSION THREAD

**Discussion by changsin** [mitmproxy/mitmproxy#6539] [General]
Is there a way of pausing and resuming ctx.master_instance? I am creating a GUI that the user can use to start and stop mitmproxy but I find that even if it's shutdown, the port number is still used by the previous instance and thus if I launch a new instance, it fails. In other words, the port number used by mitmproxy master_instance is still occupied even if shutdown is called.

So it seems that I need to either find a way of properly closing down mitmproxy (shutdown does not free up the port) or figure out how to pause and resume using the same instance.

**Comment by mhils** [comment]
Shutdown should typically free up the port, not sure what's going on here. One alternative would be to toggle the `server` option on/off.

**Reply by changsin** [reply]
No, it does not. At least when I tried it. Can you elaborate on the server option?

**Reply by mhils** [reply]
https://docs.mitmproxy.org/stable/concepts-options/#server. Simply assigning a new value to `mitmproxy.ctx.options.server` should trigger server start/stop.
